src/index.js

```diff
@@ -14,30 +14,36 @@ const tcp = new TCP()
 class MulticastDNS extends EventEmitter {
   constructor (peerInfo, options) {
     super()
-    options = options || {}
 
-    const broadcast = options.broadcast !== false
-    const interval = options.interval || (1e3 * 10) // default: 10 seconds
-    const serviceTag = options.serviceTag || '_ipfs-discovery._udp'
-    const port = options.port || 5353
-    const self = this // for event emitter
+    this.broadcast = options.broadcast !== false
+    this.interval = options.interval || (1e3 * 10)
+    this.serviceTag = options.serviceTag || '_ipfs-discovery._udp'
+    this.port = options.port || 5353
+    this.peerInfo = peerInfo
+  }
 
-    const mdns = multicastDNS({ port: port })
+  start (cb) {
```

Comment:
The ``start``method maybe should be refactored. 
Moving the ``gotQuery`` and ``gotAnswer`` to a different file/module/method, would make it more succinct.

Ideas?